pallets/pool-system/src/impls.rs

```diff
@@ -128,6 +128,8 @@ impl<T: Config> PoolMutate<T::AccountId, T::PoolId> for Pallet<T> {
 			now,
 		)?;
 
+		let mut tranches_essence_for_event = Vec::new();
```

Comment:
Instead of having this here, you can also just call `PoolDetails::essence()` below, right? Or is the reasoning that you wanna have less storage calls?

Reply:
I updated it to make it cleaner.